src/frontend/http.rs

```diff
@@ -49,14 +76,15 @@ async fn write_file_handler(pk: &str, body: BodyStream, name: Option<String>) ->
     )
     .boxed();
 
-    let Blake3Hash(hash) = write_file(pk, file_stream, name).await?;
+    let Blake3Hash(hash) = write_file(pk, file_stream, name, &is_mime_type).await?;
```

Comment:
Notice, the mime-type isn't being updated here... I don't think the async code has run yet.